### PR TITLE
align activity type with docs

### PR DESCRIPTION
**Pull request by alex-taxiera**
- Removed RichActivity since discord does not have this
- Added/fixed props found in Discord docs
  - emoji (new)
  - secrets (new)
  - party (added to)
- ~~Added props found in practice and marked which props are undocumented~~
  - ~~session_id~~
- Removed undocumented props

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -120,32 +120,32 @@ declare namespace Eris {
     url?: string;
   }
   interface Activity extends ActivityPartial<ActivityType> {
+    // start undocumented props
     id: string;
+    sync_id?: string;
+    session_id?: string;
+    // end undocumented props
```
Comment (eritbh) [inline index.d.ts]: Why do these need to be in the typings if they're undocumented?

**Review by eritbh — COMMENTED** [review]
huh, were `Activity` and `RichActivity` just never used on their own anywhere? in any case doesn't look like you did anything stupid, just curious about one thing

**Comment by alex-taxiera** [comment]
Yeah the API doesn't actually indicate rich vs non, just sometimes there are more props.

File: index.d.ts (hunk 1 above)
Comment (alex-taxiera) [reply index.d.ts]: Good point, so far I've found id is incredibly useful when used in conjunction with created_at. When you start and stop you get a fresh id with the new timestamp. Can't speak much to the other ones.

Comment (eritbh) [reply index.d.ts]: hm, what info do you get from `id` that you don't with `created_at`? I think in general we should probably avoid putting undocumented things in the typings.

Comment (alex-taxiera) [reply index.d.ts]: You could *probably* use created_at as the id, but there is technically a case where two presences are added at the same time having the same created_at. We can still remove them though and for this case I can just use some custom types.

Comment (alex-taxiera) [reply index.d.ts]: Removed in 8e1c4eb

**Review by eritbh — APPROVED** [review]

**Review by abalabahaha — APPROVED** [review]

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**